internal/deploy/kyma.go

```diff
@@ -0,0 +1,52 @@
+package deploy
+
+import (
+	"bytes"
+	"fmt"
+	"text/template"
+
+	"github.com/kyma-project/cli/internal/kube"
+)
+
+const kymaCRTemplate = `apiVersion: operator.kyma-project.io/v1alpha1
+kind: Kyma
+metadata:
+  annotations:
+    cli.kyma-project.io/source: deploy
+  labels:
+    operator.kyma-project.io/managed-by: lifecycle-manager
+  name: default-kyma
+  namespace: kcp-system
+spec:
+  channel: {{ .Channel }}
+  modules: []
+  sync:
+    enabled: {{ .Sync }}
+`
+
+// Kyma deploys the Kyma CR
+// TODO delete deploy.go when the old reconciler is gone.
+func Kyma(k8s kube.KymaKube, dryRun bool) error {
+	t, err := template.New("kymaCR").Parse(kymaCRTemplate)
```

Comment:
It would be great to also be able to pass a KymaCR-Template using a flag in the deploy command, to have it direclty configurable when deploying the setup.

Reply:
That is an interesting idea, could take a file path as flag and use that as CR (without templating).

I would nevertheless not allow to pass a go template, it would require the user to know exactly which data is available in the data structure passed to the template and all implementation details.